apps/molgenis-components/src/client/client.ts

```diff
@@ -29,18 +32,27 @@ const client: IClient = {
       deleteAllTableData: async (tableName: string) => {
         return deleteAllTableData(tableName, schemaNameCache);
       },
-      fetchSchemaMetaData: async () => {
-        schemaMetaData = await fetchSchemaMetaData(myAxios, schemaNameCache);
+      fetchSchemaMetaData: async (useCache: boolean = true) => {
+        schemaMetaData = await fetchSchemaMetaData(
+          myAxios,
+          schemaNameCache,
+          useCache
+        );
         if (!schemaNameCache) {
           schemaNameCache = schemaMetaData.name;
         }
         return deepClone(schemaMetaData);
       },
       fetchTableMetaData: async (
-        tableName: string
+        tableName: string,
+        useCache: boolean = true
       ): Promise<ITableMetaData> => {
         if (schemaMetaData === null) {
-          schemaMetaData = await fetchSchemaMetaData(myAxios, schemaNameCache);
+          schemaMetaData = await fetchSchemaMetaData(
```

Comment:
users will need to do fetchTableMetadata on other schemas. That is actually the main issue why this is so slow. I suppose that within the other client functions you are able to access the metadata cache => see convertRowToPrimaryKey (#2513) which is the main reason why I give this story priority where lookups are very slow because at every key a new client is created that retrieves schema metadata!

Reply:
that should work with this setup as consecutive requests for the schema data will keep hitting the cache